flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/sink/committables/CheckpointCommittableManagerImpl.java

```diff
@@ -158,6 +161,18 @@ CheckpointCommittableManagerImpl<CommT> copy() {
         return new CheckpointCommittableManagerImpl<>(
                 subtasksCommittableManagers.entrySet().stream()
                         .collect(Collectors.toMap(Map.Entry::getKey, (e) -> e.getValue().copy())),
+                subtaskId,
```

Comment:
Can you add a simple test that during copy, the subtaskId and numberOfSubtasks are preserved?

Reply:
Done
https://github.com/apache/flink/commit/80bef6091c3e3af18ae3e251dab694fdf7dc14eb